### DISCUSSION TITLE
Discrepancy between Mapset Verifier and lazer checker

### DISCUSSION THREAD

**Discussion by soopax** [ppy/osu#23073] [Q&A]
Osu lazer's verify tab states the slider end is unsnapped by -2.07 ms, while Mapset Verifier and AIMod state that it's only -1.811 ms. I can't determine which is correct with lazer's editor alone. (I'm on Linux and can't install stable)

Osu lazer version 2023.326.1 (latest)
Relevant modding thread:
https://osu.ppy.sh/beatmapsets/1889079/discussion/3890628/timeline#/3652484/9804213
https://osu.ppy.sh/beatmapsets/1889079/discussion/3890628/timeline#/3652484/9804273
The beatmap: [1889079 Anri - Last Surf Holiday (Night Tempo Edit).zip](https://github.com/ppy/osu/files/11129693/1889079.Anri.-.Last.Surf.Holiday.Night.Tempo.Edit.zip) 
Slider end 03:14:541 (1)

Edit: The difficulty name is "Extra" 5.48 stars

**Comment by peppy** [comment]
Does it matter? You can just fix the slider end by snapping it, yeah?

**Reply by soopax** [reply]
Sorry, but I think you're missing the point. Mapset Verifier determines this beatmap is rankable, while lazer detects the same beatmap is not rankable. This doesn't make any sense. The beatmap can't be both rankable and not rankable simultaneously. The only explanation that comes to mind is that one or the other (or both) must have a bug. Frustratingly I can't tell if the problem is coming from Mapset Verifier or your end. These tools must be accurate. If lazer is wrong about this then it might mean that creating perfectly timed beatmaps is impossible in lazer.

**Reply by peppy** [reply]
I understand that, but both 1.8 and 2ms are quite far unsnapped, so my immediate thought would be "fix it regardless".

There can be floating point differences in sliders which could lead to this margin of error on a very fast slider, so I'm not sure how much attention we need to pay to it if it's within error margin.

**Reply by soopax** [reply]
> so my immediate thought would be "fix it regardless"

Assuming you mean the beatmap, I don't think the creator can fix it anymore because it was ranked already (I could be mistaken). Should someone reopen the issue on the beatmap discussion page?
> There can be floating point differences in sliders which could lead to this margin of error on a very fast slider, so I'm not sure how much attention we need to pay to it if it's within error margin.

Ok

**Reply by peppy** [reply]
FWIW I'm still looking into this, but on both stable and lazer the difference is large enough that re-snapping the slider will cause it to get shorter (ie. whatever created the slider, probably an external app, stated the length was longer than the actual path length). This is not considered a breaking issue, but generally cannot happen if you use the lazer/stable editor, so should be an edge case.
